### PR TITLE
[#134990] facility dash redirect

### PR DESCRIPTION
**Pull request by meara**
- Refactor logic for the "default" facility admin homepage path to allow it to be extended/overridden by school-specific engines
- https://pm.tablexi.com/issues/134990
- Accompanies https://github.com/tablexi/nucore-uic/pull/130

### CODE REVIEW

**Comment by meara** [comment]
@jhanggi updated

**Comment by jhanggi** [comment]
Well, looks like we still have a couple test failures looking for old behavior.

**Comment by jhanggi** [comment]
👍

File: lib/ability.rb [hunk 1]
```diff
@@ -148,7 +148,7 @@ def initialize(user, resource, controller = nil)
 
         can :index, User if controller.is_a?(FacilityUserReservationsController)
 
-        can [:list, :show], Facility
+        can [:list, :dashboard, :show], Facility
```
Comment (meara) [inline lib/ability.rb]: @jhanggi, I just wanted to double check--I added this to the ability file to resolve the sanger spec failures. This permission is correctly located, right? It shouldn't be overreaching anywhere?

Comment (jhanggi) [inline lib/ability.rb]: Just curious, without this, what happens if you're a lowly facility staff and you try to get in to the manage facility? I would expect it to fail. Maybe we should add something in a feature spec?

**Pull request merged**